### PR TITLE
Load Current Map button added

### PR DESCRIPTION
**Pull request by BannZay**
Should be good feature for testing/mapping
Was suggested on [ddnet forum](https://forum.ddnet.tw/viewtopic.php?f=48&t=4121)

### CODE REVIEW

**Comment by def-** [comment]
That only works if you didn't originally have the map. You can't just use `downloadedmaps/foo_bar.map`. There are two clean approaches to solve this imo:

1. Save the exact map path that was used for opening each map, then load this location. Note that this can be a pretty long string.
2. Look at the map name and CRC, which are available already, then run a search to find a map that matches those. Something similar happens for regular game map opening already.

**Comment by BannZay** [comment]
I guess better to store instead of searching among all maps

**Comment by def-** [comment]
Not necessarily. It's possible the map file has already been deleted, but another copy is still lying around somewhere. But that's a corner case that won't matter much I believe.

**Comment by BannZay** [comment]
But this copy is not exactly "Current" map, right? Even if exists some copy even with the same crc. Downloading for editing this copy could brake plans, folowing which this copy was maded.

One more thing is that onDisconnect mapPath wont reset but its even cool in some cases. 
So for now it works like "Open last loaded ingame map". 
Yours thought about this?

**Comment by timakro** [comment]
I would prefer being able to load the last map even if I'm not on a server.

**Comment by def-** [comment]
I don't see the big problem with just opening any copy with same crc, that's the same approach regular map opening takes. I don't think you need ot reset the map on disconnect.

**Comment by BannZay** [comment]
Searching approach will win in very rarely situation... I cant even call it "win", cause  after some manipulations in file system you cant be sure(which is important) what exactly will do this feature. (If you want to use "save" instead "save as" in future)
All thing about I`m saying important for mappers, and for whom this feature made. You may not understand me if you have never lose a lot of your work with mapping
> I don't see the big problem

I will agree. Maybe I`m wrong in some cases, but I still prefer much safe behaviour

**Comment by def-** [comment]
I think this is a good new feature, please fix it to remember the full path and it will be merged.

**Comment by BannZay** [comment]
>  please fix it
I thought that it was over. I might say sorry about force push.
Or there is still some troubles?

File: src/engine/client/client.h [hunk 1]
```diff
@@ -116,6 +116,7 @@ class CClient : public IClient, public CDemoPlayer::IListener
 
 	//
 	char m_aCurrentMap[256];
+	char m_aCurrentMapPath[512];
```
Comment (def-) [inline src/engine/client/client.h]: Use MAX_PATH_LENGTH instead magic number 512.

**Comment by def-** [comment]
Oh, I totally missed that. Just a small fix and it can be merged.

**Pull request merged**